src/modules/bootloader/main.py

```diff
@@ -145,43 +127,84 @@ def create_systemd_boot_conf(install_path, efi_dir, uuid, entry, entry_name, ker
         kernel_params.append("resume=/dev/mapper/{!s}".format(
             swap_outer_mappername))
 
-    kernel_line = get_kernel_line(kernel_type)
+    kernel_line = entry_name + " " + kernel_version
     libcalamares.utils.debug("Configure: \"{!s}\"".format(kernel_line))
 
     if kernel_type == "fallback":
-        img = libcalamares.job.configuration["fallback"]
-        entry_name = entry_name + "-fallback"
+        version_string = kernel_version + "-fallback"
+        initrd = "initrd-fallback"
+        mkinitcpio_option = "-S autodetect"
     else:
-        img = libcalamares.job.configuration["img"]
+        version_string = kernel_version
+        initrd = "initrd"
+        mkinitcpio_option = ""
 
-    conf_path = os.path.join(install_path + efi_dir,
-                             "loader",
-                             "entries",
-                             entry_name + ".conf")
+    # get the machine-id
+    with open(os.path.join(install_path, "etc", "machine-id"), 'r') as machineid_file:
+        machine_id = machineid_file.read().rstrip('\n')
 
-    # Copy kernel and initramfs to a subdirectory of /efi partition
-    files_dir = os.path.join(install_path + efi_dir, entry_name)
-    os.mkdir(files_dir)
+    # Copy kernel to a subdirectory of /efi partition
+    machine_dir = os.path.join(install_path + efi_dir, machine_id)
+    try:
+        os.mkdir(machine_dir)
+    except FileExistsError: # We can ignore errors caused by the directory existing already
+        pass
 
-    kernel_path = install_path + kernel
+    files_dir = os.path.join(machine_dir, kernel_version)
+    try:
+        os.mkdir(files_dir)
+    except FileExistsError: # We can ignore errors caused by the directory existing already
+        pass
+    
+    kernel_path = os.path.join(install_path, kernel)
     kernel_name = os.path.basename(kernel_path)
-    shutil.copyfile(kernel_path, os.path.join(files_dir, kernel_name))
-
-    img_path = install_path + img
-    img_name = os.path.basename(img_path)
-    shutil.copyfile(img_path, os.path.join(files_dir, img_name))
+    shutil.copyfile(kernel_path, os.path.join(files_dir, "linux"))
 
+    # generate the initramfs - this is Arch specific and should be replaced for other distros
+    try:
+        subprocess.run(["chroot " +
+                        install_path + 
+                        " mkinitcpio -k " + kernel_version + " -g " + os.path.join("/", os.path.relpath(files_dir, install_path), initrd)],
```

Comment:
Yes, that isn't documented very well.  I sort of assumed it would need to be refactored since it is overly specific.

To be clear, the `relpath` is producing a path suitable for use in the chroot by removing `install_path`.  I think it could be made self documenting by making the relpath call a separate statement with a descriptive variable name.